pkg/localstore/migration.go

```diff
@@ -101,3 +105,264 @@ func getMigrations(currentSchema, targetSchema string, allSchemeMigrations []mig
+			tag := make([]byte, 4)
+			binary.BigEndian.PutUint32(tag, fields.Tag)
+			return tag, nil
+		},
+		DecodeValue: func(keyItem shed.Item, value []byte) (e shed.Item, err error) {
+			if len(value) == 4 { // only values with tag should be decoded
+				e.Tag = binary.BigEndian.Uint32(value)
+			}
+			return e, nil
+		},
+	})
+	if err != nil {
+		return err
+	}
+	gcIndex, err := db.shed.NewIndex("AccessTimestamp|BinID|Hash->nil", shed.IndexFuncs{
+		EncodeKey: func(fields shed.Item) (key []byte, err error) {
+			b := make([]byte, 16, 16+len(fields.Address))
+			binary.BigEndian.PutUint64(b[:8], uint64(fields.AccessTimestamp))
+			binary.BigEndian.PutUint64(b[8:16], fields.BinID)
+			key = append(b, fields.Address...)
+			return key, nil
+		},
+		DecodeKey: func(key []byte) (e shed.Item, err error) {
+			e.AccessTimestamp = int64(binary.BigEndian.Uint64(key[:8]))
+			e.BinID = binary.BigEndian.Uint64(key[8:16])
+			e.Address = key[16:]
+			return e, nil
+		},
+		EncodeValue: func(fields shed.Item) (value []byte, err error) {
+			return nil, nil
+		},
+		DecodeValue: func(keyItem shed.Item, value []byte) (e shed.Item, err error) {
+			return e, nil
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Create a index structure for excluding pinned chunks from gcIndex
+	gcExcludeIndex, err := db.shed.NewIndex("Hash->nil", shed.IndexFuncs{
+		EncodeKey: func(fields shed.Item) (key []byte, err error) {
+			return fields.Address, nil
+		},
+		DecodeKey: func(key []byte) (e shed.Item, err error) {
+			e.Address = key
+			return e, nil
+		},
+		EncodeValue: func(fields shed.Item) (value []byte, err error) {
+			return nil, nil
+		},
+		DecodeValue: func(keyItem shed.Item, value []byte) (e shed.Item, err error) {
+			return e, nil
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	var lim = 10000
+	count := 0
+
+	truncate := func(i shed.Index) error {
+		batch := new(leveldb.Batch)
+
+		err := i.Iterate(func(item shed.Item) (stop bool, err error) {
+			if err = i.DeleteInBatch(batch, item); err != nil {
+				return true, err
+			}
+			count++
+			if count%lim == 0 {
+				db.logger.Infof("truncate writing batch. processed %d", count)
+				err := db.shed.WriteBatch(batch)
+				if err != nil {
+					return true, err
+				}
+				batch = new(leveldb.Batch)
+			}
+			return false, nil
+		}, nil)
+		if err != nil {
+			return err
+		}
+		return db.shed.WriteBatch(batch)
+	}
+	start := time.Now()
+	db.logger.Info("truncating indexes")
+
+	for _, v := range []struct {
+		name string
+		idx  shed.Index
+	}{
+		{"pullsync", pullIndex},
+		{"pushsync", pushIndex},
+		{"gc", gcIndex},
+		{"gcExclude", gcExcludeIndex},
+		{"retrievalAccess", retrievalAccessIndex},
+		{"retrievalData", retrievalDataIndex},
+	} {
+		db.logger.Infof("truncating %s index", v.name)
+		err := truncate(v.idx)
+		if err != nil {
+			return fmt.Errorf("truncate %s index: %w", v.name, err)
+		}
+		db.logger.Infof("truncated %d %s index entries", v.name, count)
```

Comment:
I see on my terminal
```
INFO[2021-05-17T16:40:05+02:00] truncated %!d(string=pullsync) %!s(int=2) index entries 
```

Reply:
Besides, I suggest to put all log lines to `debug` and only have two info log line saying that `migration is in process` and `migration successful`